app/services/offer_validations.rb

```diff
@@ -9,12 +9,19 @@ class OfferValidations
   attr_accessor :application_choice, :course_option, :conditions
 
   validates :course_option, presence: true
+  validate :current_recruitment_cycle_year, if: %i[application_choice course_option]
   validate :conditions_count, if: :conditions
   validate :conditions_length, if: :conditions
   validate :identical_to_existing_offer?, if: %i[application_choice course_option]
   validate :ratifying_provider_changed?, if: %i[application_choice course_option]
   validate :restrict_reverting_rejection, if: :application_choice
 
+  def current_recruitment_cycle_year
+    if course_option.course.recruitment_cycle_year != application_choice.current_recruitment_cycle_year
```

Comment:
👍  validation living in offer validations! However if I try to make an offer on an application from the previous cycle which was rejected, without changing the course I am still able to as the course year and application year are similar. Maybe if we just check for the `application_choice.current_recruitment_cycle_year` being in the current cycle we capture both making an offer to the existing course (for an application that was rejected in the previous cycle), and if they attempt to change the course that is also caught? let me know your thoughts and if I missed anything

Reply:
The issue I came across with anything to do with enforcing the current cycle check was that we could no longer generate test applications for past or future cycles. Because of the way `OfferValidations` is initialized in `MakeOffer` and `ChangeOffer` it's not accessible to the caller otherwise we could say _skip current cycle validation_ in the case of test applications.
We could inject the validations and configure them to conditionally call the current cycle validation, **or**  set an instance var on `application_choice` or `course_option` which signifies we skip the validation (a bit icky/indirect) **or** move the validation elsewhere (which also seems sub-optimal).